paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationInterceptor.kt

```diff
@@ -300,7 +308,16 @@ internal class DefaultIntentConfirmationInterceptor @Inject constructor(
             if (intent.isConfirmed) {
                 NextStep.Complete(isForceSuccess = false)
             } else if (intent.requiresAction()) {
-                NextStep.HandleNextAction(clientSecret)
+                val attachedPaymentMethodId = intent.paymentMethodId
+
+                if (attachedPaymentMethodId != null && attachedPaymentMethodId != paymentMethod.id) {
+                    NextStep.Fail(
+                        cause = InvalidDeferredIntentUsageException(),
+                        message = resolvableString(R.string.stripe_paymentsheet_invalid_deferred_intent_usage),
```

Comment:
Interesting that this is the way to represent failures/errors.

Like on iOS, failure/errors are ~always represented by something that conforms to `Error`.  And a method that fails either vends the error by `throw`ing it, returning it, etc.

This is nice b/c it means errors are all the same and can be handled uniformly. There is no code like this to transform error-like objects from one to another:
```
                        type = if (nextStep.cause is InvalidDeferredIntentUsageException) {
                            PaymentConfirmationErrorType.MerchantIntegration.InvalidDeferredIntentUsage
                        } else {
                            PaymentConfirmationErrorType.Internal
                        },
```

Reply:
Yeah this was modeled this way because we were handling error cases differently for some reason on `PaymentSheet`. I just had gotten around to fixing those error issues for the intent confirmation flow.

There's a path to make this more uniform though by making this exception a `StripeException` which can do something similar.